### ISSUE TITLE
derivedData capability not respected with new appium-webdriveragent

### ISSUE THREAD

**Issue by dpgraham** [appium/appium#12836]
## The problem

When I set these capabilities: 

```javascript
{
  "platformName": "iOS",
  "automationName": "XCUITest",
  "deviceName": "iPhone SE",
  "platformVersion": "12.0",
  "showIOSLog": false,
  "showXcodeLog": true,
  "browserName": "Safari",
  "noReset": false,
  "waitForQuiescence": false,
  "fullReset": false,
  "usePrebuiltWDA": true,
  "derivedDataPath": "/Users/danielgraham/appium/bundles/WDA/DerivedData/WebDriverAgent",
  "bootstrapPath": "/Users/danielgraham/appium/bundles/WDA/WebDriverAgent"
}
```

It doesn't respect `derivedDataPath`. It installs the Carthage dependencies for it even though the dependencies are already there.

This is the pull request that caused the issue

https://github.com/appium/appium-xcuitest-driver/pull/980

It happened between XCUI 2.116.3 ~> 2.117.0 (2.116.3 works fine)

This is the checkDependencies code for 2.116.3: https://github.com/appium/appium-xcuitest-driver/blob/v2.116.3/lib/wda/utils.js#L66

This is the checkDependencies code for 2.117.0: https://github.com/appium/WebDriverAgent/blob/master/index.js#L114

The code in appium-xcuitest-driver@2.116.3 checks dependencies and checks for Carthage

The code in WebDriverAgent simply fetches the dependencies without doing a check

The code at https://github.com/appium/appium-xcuitest-driver/blob/v2.116.3/lib/wda/utils.js#L66 should probably just be calling `fetchDependencies` from `appium-webdriveragent` if it can't find Carthage dependencies.

If someone makes a PR for this, ping me and I can test out the fix.

## Environment

* Appium version (or git revision) that exhibits the issue: XCUITestDriver 1.117.0
* Last Appium version that did not exhibit the issue (if applicable): XCUITestDriver 1.116.3
* Desktop OS/version used to run Appium: Mac OSX Mojave
* Node.js version (unless using Appium.app|exe): 12
* Npm or Yarn package manager: NPM
* Mobile platform/version under test: IOS
* Real device or emulator/simulator: Simulator
* Appium CLI or Appium.app|exe: CLI

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs

Snippet of logs

```
[debug] [IDB] Connecting idb service to 'ABD74AD0-9878-4672-B0D1-4CF5A5AF9556'
[XCUITest] idb will not be used for Simulator interaction. Original error: 'idb_companion' has not been found in PATH. Is it installed? Read https://www.fbidb.io for more details
[debug] [BaseDriver] Event 'simStarted' logged at 1561592739402 (16:45:39 GMT-0700 (Pacific Daylight Time))
[debug] [XCUITest] Starting log capture for iOS Simulator with udid 'ABD74AD0-9878-4672-B0D1-4CF5A5AF9556', using 'xcrun simctl spawn ABD74AD0-9878-4672-B0D1-4CF5A5AF9556 log stream --style compact'
[debug] [BaseDriver] Event 'logCaptureStarted' logged at 1561592747151 (16:45:47 GMT-0700 (Pacific Daylight Time))
[XCUITest] Using WDA path: '/Users/danielgraham/appium/bundles/WDA/WebDriverAgent'
[XCUITest] Using WDA agent: '/Users/danielgraham/appium/bundles/WDA/WebDriverAgent/WebDriverAgent.xcodeproj'
[debug] [XCUITest] No obsolete cached processes from previous WDA sessions listening on port 8100 have been found
[debug] [XCUITest] Starting WebDriverAgent initialization with the synchronization key '/Users/danielgraham/appium/bundles/WDA/DerivedData/WebDriverAgent'
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[WD Proxy] Got an unexpected response: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[debug] [XCUITest] WDA is not listening at 'http://localhost:8100/'
[debug] [XCUITest] WDA is currently not running. There is nothing to cache
[debug] [XCUITest] Trying to start WebDriverAgent 2 times with 10000ms interval
[debug] [XCUITest] These values can be customized by changing wdaStartupRetries/wdaStartupRetryInterval capabilities
[debug] [BaseDriver] Event 'wdaStartAttempted' logged at 1561592760053 (16:46:00 GMT-0700 (Pacific Daylight Time))
[XCUITest] Launching WebDriverAgent on the device
[WebDriverAgent] Fetching dependencies
[debug] [WebDriverAgent] tvOS platform will not be included into Carthage bootstrap, because no Simulator devices have been created for it
[WebDriverAgent] Installing/updating dependencies for platforms 'iOS'
^C[Appium] Received SIGINT - shutting down
```

It shouldn't be reaching `Fetching dependencies` part of the log

**Comment by dpgraham** [comment]
cc @imurchie @mykola-mokhnach @umutuzgur ^

**Comment by KazuCocoa** [comment]
> The code in WebDriverAgent simply fetches the dependencies without doing a check

If the carthage dir exists, https://github.com/appium/WebDriverAgent/blob/master/index.js#L77 returns false just after `Fetching dependencies` message without calling carthage command.
The message issue...?

**Comment by dpgraham** [comment]
I think this needs to be changed: https://github.com/appium/WebDriverAgent/blob/master/index.js#L75

If the `usePrebuiltWDA` flag is set to `true` then don't install carthage dependencies. Leave it up to the user to make sure that the carthage deps are correct.

**Comment by KazuCocoa** [comment]
note:
https://github.com/appium/appium-xcuitest-driver/pull/992 (merged) and https://github.com/appium/WebDriverAgent/pull/182 (not merged yet) also necessary to fix issues in appium/appium-xcuitest-driver#980

**Comment by KazuCocoa** [comment]
everything has been merged. https://github.com/appium/appium/issues/12836#issuecomment-506676336
